### PR TITLE
Add lower and upper arguments to Array

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/parametrization/data.py [hunk 1]
```diff
@@ -84,39 +84,68 @@ class Data(core.Parameter):
...
+            layer = _datalayers.Bound(
+                lower=lower, upper=upper, uniform_sampling=init is None and num_bounds == 2
+            )
+            if num_bounds == 2:
+                sigma_val = (layer.bounds[1] - layer.bounds[0]) / 6
+        # set parameters
+        if not mutable_sigma:
+            sigma: tp.Any = sigma_val[0] if sigma_val.size == 1 else sigma_val
+        elif sigma_val.size == 1:
+            sigma = Log(init=sigma_val[0], exponent=2.0, mutable_sigma=False)
```
Comment (jrapin) [inline nevergrad/parametrization/data.py:142]: @teytaud important question here, should sigma be log-distributed? or linearly-distributed but bounded by (unreachable) 0?
My feeling is that Log seems to evolve too quickly and be unstable, then again, I'm not that sure (eg: TBPSA is using logarithmic updates right?)

Comment (teytaud) [reply nevergrad/parametrization/data.py:142]: From all the theory of Anne regarding Markov chains and the experimental works by Rechenberg and others, I would say yes, go for log.

Comment (jrapin) [reply nevergrad/parametrization/data.py:142]: from @teytaud coeff to be updated, but still log indeed

File: nevergrad/optimization/test_optimizerlib.py [hunk 2]
```diff
@@ -380,7 +380,7 @@ def test_optimization_doc_parametrization_example() -> None:
     assert len(recom.args) == 1
     testing.assert_set_equal(recom.kwargs, ["y"])
     value = _square(*recom.args, **recom.kwargs)
-    assert value < 0.2  # should be large enough by an order of magnitude
+    assert value < 0.25  # should be large enough by an order of magnitude (but is not :s)
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:383]: @teytaud I've had to increase this tolerance due to flakiness :s

File: nevergrad/parametrization/data.py [hunk 3]
```diff
@@ -88,39 +88,75 @@ class Data(core.Parameter):
...
+        # set parameters
+        sigma = sigma[0] if sigma.size == 1 else sigma
+        if mutable_sigma:
+            siginit = sigma
+            # for the choice of the base:
+            # cf Evolution Strategies, Hans-Georg Beyer (2007)
+            # http://www.scholarpedia.org/article/Evolution_strategies
+            # we want:
+            # sigma *= exp(gaussian / sqrt(dim))
+            base = float(np.exp(1.0 / np.sqrt(2 * init.size)))
```
Comment (jrapin) [inline nevergrad/parametrization/data.py:151]: @teytaud this should work

**Pull request merged**